### PR TITLE
Super admin

### PR DESCRIPTION
**Pull request by princeBacancy**

### CODE REVIEW

File: app/controllers/appointments_controller.rb [hunk 1]
```diff
@@ -1,33 +1,71 @@
...
     appointment = Appointment.new appointment_params
     if appointment.save
       flash[:errors] = 'Your appointment has been boooked'
       notification = Notification.create(recipient_id: appointment.patient_id,
-                                         body: "your appointment for dr.#{appointment.doctor.name} on #{appointment.date} has been booked")
+                                         body: "your appointment for dr.
+                                         #{appointment.doctor.name} on
+                                         #{appointment.date} has been booked")
       ActionCable.server.broadcast('notification_channel',
                                    notification: notification)
```
Comment (jinal-Bacancy) [inline app/controllers/appointments_controller.rb:20]: This code is common to create, update and cancel-appointment method: only half message is different, make one private method and pass message and object in that. And call that method in these three methods.

File: app/controllers/hospitals_controller.rb [hunk 2]
```diff
@@ -43,13 +40,11 @@ def create
   end
 
   def edit
-    @hospital = Hospital.find(params[:id])
-    authorize @hospital
+    @hospital = authorize Hospital.find(params[:id])
```
Comment (jinal-Bacancy) [inline app/controllers/hospitals_controller.rb:43]: make one method find_hospital and call them in the callback, before edit, update and delete methods.

File: app/controllers/appointments_controller.rb (hunk 1 above)
Comment (princeBacancy) [reply app/controllers/appointments_controller.rb]: done

File: app/controllers/hospitals_controller.rb (hunk 2 above)
Comment (princeBacancy) [reply app/controllers/hospitals_controller.rb]: done

File: app/controllers/doctors_controller.rb [hunk 3]
```diff
@@ -1,79 +1,75 @@
...
-    @doctor = Doctor.new
-    authorize @doctor
+    @doctor = authorize Doctor.new
     @availability = @doctor.build_availability
   end
 
   def edit
-    @doctor = Doctor.find_by(id: params[:id])
-    authorize @doctor
+    @doctor = authorize Doctor.find_by(id: params[:id])
```
Comment (jinal-Bacancy) [inline app/controllers/doctors_controller.rb:23]: Remove this line, add this method in doctor_find callback

Comment (princeBacancy) [reply app/controllers/doctors_controller.rb:23]: ae function ma atyare policy nathi create kari aetle authorize vado code chhe ae function ma so ae pachhi kari dais policy add karine to chalse k atyare j kari dav

Comment (jinal-Bacancy) [reply app/controllers/doctors_controller.rb:23]: Okay, it will work, for now, correct it later.

Comment (princeBacancy) [reply app/controllers/doctors_controller.rb:23]: done

**Pull request merged**